### PR TITLE
SAK-45854: Discussions > Statistics > Student View > Clicking any link in the "Subject" column returns an empty page

### PR DESCRIPTION
**Pull request by stetsche**
https://jira.sakaiproject.org/browse/SAK-45854

Make to discussion posts in statistics render for the author (student)

### CODE REVIEW

File: msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/MessageForumStatisticsBean.java [hunk 1]
```diff
@@ -2184,6 +2185,21 @@ private String getExternalParameterByKey(String parameterId)
 		return (String) paramMap.get(parameterId);    
 	}
 
+	public boolean getIsAuthor() {
+		String siteUserId;
+		String currentUserId;
+		try
+		{
+			siteUserId = userDirectoryService.getUser(selectedSiteUserId).getId();
```
Comment (bgarciaentornos) [inline msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/MessageForumStatisticsBean.java:2193]: I think there's something wrong in here... If the 'selectedSiteUserId' is the eid then you should call getUserByEid, and if it's not then you can directly use that id in the comparison.

**Review by bgarciaentornos — CHANGES_REQUESTED** [review]

**Review by bjones86 — APPROVED** [review]

File: msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/MessageForumStatisticsBean.java [hunk 2]
```diff
@@ -2184,6 +2184,9 @@ private String getExternalParameterByKey(String parameterId)
 		return (String) paramMap.get(parameterId);    
 	}
 
+	public boolean getIsAuthor() {
+		return selectedSiteUserId == userDirectoryService.getCurrentUser().getId();
```
Comment (ottenhoff) [inline msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/MessageForumStatisticsBean.java:2188]: Is this comparing strings?!?! Is this okay?

Comment (mpellicer) [reply msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/MessageForumStatisticsBean.java:2188]: I think it is, I kinda don't like it but I think it works fine.

Comment (ottenhoff) [reply msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/MessageForumStatisticsBean.java:2188]: "comparing the reference of the given strings, depending on if they are referring to the same objects." ... how can they refer to the same object?

Comment (mpellicer) [reply msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/MessageForumStatisticsBean.java:2188]: Looks like it compares the String value. Is this causing an issue in any instance?

Comment (ottenhoff) [reply msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/MessageForumStatisticsBean.java:2188]: Sorry this was fixed in SAK-45854 ef3817fd098e396ae722425cb3884f4da737667d

**Pull request merged**